src/software/simulation/convert_primitive_to_nanopb.cpp

```diff
@@ -0,0 +1,22 @@
+#include "software/simulation/convert_primitive_to_nanopb.h"
+
+#include <pb_decode.h>
+
+#include "software/simulation/serialize_primitive_to_proto.h"
+
+PrimitiveMsg convertPrimitiveToNanoPb(const Primitive& primitive)
```

Comment:
```suggestion
PrimitiveMsg createNanoPbPrimitiveMsg(const Primitive& primitive)
```
to be consistent with proto/message_translation naming

Reply:
Changed as suggested. Left the file and target name as-is because while the types convey what input to the conversion is, the file/target name do not.